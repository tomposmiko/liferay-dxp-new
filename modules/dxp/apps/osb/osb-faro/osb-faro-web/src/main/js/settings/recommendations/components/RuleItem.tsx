import MetadataTag from './MetadataTag';
import React from 'react';
import {getFilterValueBreakdown, RULE_NAME_LABEL_MAP} from '../utils/utils';

interface IRuleItemProps {
	name: string;
	value: string;
}

const RuleItem: React.FC<IRuleItemProps> = ({name, value}) => {
	const {exactMatchSign, metadataTag, rule} = getFilterValueBreakdown(value);

	const exactMatch = exactMatchSign === '=';

	return (
		<div className='rule-item-root d-flex align-items-baseline'>
			<b>{`${RULE_NAME_LABEL_MAP[name]}:`}</b>

			{metadataTag && <MetadataTag value={metadataTag} />}

			<span className='rule-value'>
				{exactMatch ? `"${rule}"` : rule}
			</span>
		</div>
	);
};

export default RuleItem;
