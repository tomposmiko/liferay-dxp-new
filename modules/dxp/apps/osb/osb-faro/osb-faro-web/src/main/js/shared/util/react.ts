import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
