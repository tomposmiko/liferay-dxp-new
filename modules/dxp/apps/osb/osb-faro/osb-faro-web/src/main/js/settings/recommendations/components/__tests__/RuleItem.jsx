import React from 'react';
import RuleItem from '../RuleItem';
import {render} from '@testing-library/react';

jest.unmock('react-dom');

describe('RuleItem', () => {
	it('should render', () => {
		const {container} = render(
			<RuleItem
				name='includeFilter'
				value='og:url = https://www.liferay.com'
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
