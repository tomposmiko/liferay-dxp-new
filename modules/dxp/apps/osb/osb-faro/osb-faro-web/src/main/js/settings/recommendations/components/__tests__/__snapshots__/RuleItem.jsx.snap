// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RuleItem should render 1`] = `
<div>
  <div
    class="rule-item-root d-flex align-items-baseline"
  >
    <b>
      Include:
    </b>
    <div
      class="label label-root metadata-tag-root label-lg"
    >
      <span
        class="label-item"
      >
        og:url
      </span>
    </div>
    <span
      class="rule-value"
    >
      "https://www.liferay.com"
    </span>
  </div>
</div>
`;
